core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -602,6 +604,21 @@ private[spark] class ExecutorAllocationManager(
     // place the executors.
     private val stageIdToExecutorPlacementHints = new mutable.HashMap[Int, (Int, Map[String, Int])]
 
+    override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+      val jobGroupId = if (jobStart.properties != null) {
+        jobStart.properties.getProperty(SparkContext.SPARK_JOB_GROUP_ID)
+      } else {
+        ""
+      }
+      val maxConcurrentTasks = conf.getInt(s"spark.job.$jobGroupId.maxConcurrentTasks",
+        Int.MaxValue)
+
+      logInfo(s"Setting maximum concurrent tasks for group: ${jobGroupId} to $maxConcurrentTasks")
+      allocationManager.synchronized {
+        allocationManager.maxConcurrentTasks = maxConcurrentTasks
```

Comment:
if you have multiple jobs running concurrently, you are not keeping a separate maxConcurrentTasks for each one.